# Libraries and File

In [49]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [50]:
df= pd.read_csv('/content/DFORE350.csv', low_memory=False)

In [51]:
df.shape

(102034, 40)

In [52]:
df.isnull().sum()[df.isnull().sum()>0]

location_id                    137
Location_Name                  139
Branch_Id                     2964
Branch_Name                   2964
state                          328
Daily_Max_Room_Vend_Hours        2
Total_Vend_Hours                 2
Overall_Utilization              2
Daily_Peak_Utilization           2
Full_Room_Util_Gross_Hours    1596
Full_Room_Util_Vend_PCT       1596
Full_Room_Util_Total_PCT      1596
AvgUtil_T                        1
AvgUtil_Sa                       1
nrpm                          3299
ContractExpirationDate        1510
Days_To_ContractExpiration    1510
dtype: int64

In [53]:
df.dropna(inplace=True)

In [54]:
df.shape

(94830, 40)

In [55]:
df.dtypes

Reporting_level                object
vendor                         object
as400id                        object
location_id                    object
Location_Name                  object
Branch_Id                     float64
Branch_Name                    object
state                          object
room_id                        object
machine_type                   object
TimeSinceFirstVend              int64
Tenure_Flag                    object
Txn_Avg_Machine_Cnt             int64
Room_Open_Hours                 int64
Daily_Max_Room_Vend_Hours     float64
Base_Vends                      int64
Base_Vend_Amount              float64
Top_Up_Vends                    int64
Top_Up_Vend_Amount            float64
Total_Vend_Hours              float64
Total_Hours_Open                int64
Overall_Utilization           float64
Max_Daily_Vends                 int64
Daily_Peak_Utilization        float64
Full_Room_Util_Gross_Hours    float64
Full_Room_Util_Vend_PCT       float64
Full_Room_Ut

In [56]:
df['machine_type']=df.machine_type.str.upper()

# Rooms & MAchines

In [57]:
df.groupby('vendor').room_id.nunique()

vendor
CSCGo    16406
DI       32336
Name: room_id, dtype: int64

In [58]:
pd.crosstab(df.vendor, df.machine_type, df.Txn_Avg_Machine_Cnt, aggfunc='sum')

machine_type    DRYER  STACKED   WASHER
vendor                                 
CSCGo         37179.0      NaN  37657.0
DI            72087.0   1071.0  75308.0

In [59]:
df1= df.loc[df.Txn_Avg_Machine_Cnt!=1]

In [60]:
df1.groupby('vendor').room_id.nunique()

vendor
CSCGo    12109
DI       21269
Name: room_id, dtype: int64

In [61]:
pd.crosstab(df1.vendor, df1.machine_type, df1.Txn_Avg_Machine_Cnt, aggfunc='sum')

machine_type    DRYER  STACKED   WASHER
vendor                                 
CSCGo         32734.0      NaN  33045.0
DI            61232.0    549.0  64276.0

## Floor and Cap

In [62]:
df1.dtypes[df1.dtypes!='object'].index

Index(['Branch_Id', 'TimeSinceFirstVend', 'Txn_Avg_Machine_Cnt',
       'Room_Open_Hours', 'Daily_Max_Room_Vend_Hours', 'Base_Vends',
       'Base_Vend_Amount', 'Top_Up_Vends', 'Top_Up_Vend_Amount',
       'Total_Vend_Hours', 'Total_Hours_Open', 'Overall_Utilization',
       'Max_Daily_Vends', 'Daily_Peak_Utilization',
       'Full_Room_Util_Gross_Hours', 'Full_Room_Util_Vend_PCT',
       'Full_Room_Util_Total_PCT', 'Avg_Daily_Vends', 'VDPM', 'Vend_Price',
       'AvgUtil_M', 'AvgUtil_T', 'AvgUtil_W', 'AvgUtil_Th', 'AvgUtil_F',
       'AvgUtil_Sa', 'AvgUtil_Su', 'nrpm', 'Days_To_ContractExpiration'],
      dtype='object')

In [80]:
sums= ['Txn_Avg_Machine_Cnt','Base_Vends', 'Base_Vend_Amount', 'Top_Up_Vends', 'Top_Up_Vend_Amount', 'Total_Vend_Hours', 'Total_Hours_Open']
utils1= ['Overall_Utilization','Daily_Peak_Utilization','Full_Room_Util_Total_PCT']
utils= ['Overall_Utilization','Daily_Peak_Utilization','Full_Room_Util_Vend_PCT','Full_Room_Util_Total_PCT', 'Avg_Daily_Vends', 'VDPM', 'Vend_Price']
weeklyutil= ['AvgUtil_M', 'AvgUtil_T', 'AvgUtil_W', 'AvgUtil_Th', 'AvgUtil_F', 'AvgUtil_Sa', 'AvgUtil_Su']

In [63]:
otl = ['Txn_Avg_Machine_Cnt',
       'Room_Open_Hours', 'Daily_Max_Room_Vend_Hours', 'Base_Vends',
       'Base_Vend_Amount', 'Top_Up_Vends', 'Top_Up_Vend_Amount',
       'Total_Vend_Hours', 'Total_Hours_Open', 'Overall_Utilization',
       'Max_Daily_Vends', 'Daily_Peak_Utilization',
       'Full_Room_Util_Gross_Hours', 'Full_Room_Util_Vend_PCT',
       'Full_Room_Util_Total_PCT', 'Avg_Daily_Vends', 'VDPM', 'Vend_Price',
       'AvgUtil_M', 'AvgUtil_T', 'AvgUtil_W', 'AvgUtil_Th', 'AvgUtil_F',
       'AvgUtil_Sa', 'AvgUtil_Su', 'nrpm']

In [85]:
df1.groupby(['vendor', 'machine_type'])[weeklyutil].describe().T

vendor                   CSCGo                ...          DI              
machine_type             DRYER        WASHER  ...     STACKED        WASHER
AvgUtil_M  count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       6.395628      5.183466  ...    9.296652      5.907880
           std        4.547027      3.879055  ...    4.825155      3.291533
           min        0.000000      0.000000  ...    0.000000      0.000000
           25%        3.197825      2.503700  ...    5.811575      3.624700
           50%        5.666450      4.493650  ...    8.291950      5.367900
           75%        8.728600      7.007575  ...   12.280150      7.490300
           max       71.250000     45.050500  ...   25.389900     34.722200
AvgUtil_T  count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       5.379767      4.340997  ...    8.231278      5.123236
           std        3.816129      3.124715  ...    4.295676      2.882841
           min        0.000000      0.000000  ...    0.000000      0.000000
           25%        2.726450      2.182100  ...    5.008125      3.162300
           50%        4.819100      3.891350  ...    7.751650      4.631100
           75%        7.299900      5.890325  ...   10.797900      6.543000
           max       58.333300     31.666600  ...   22.754500     36.380800
AvgUtil_W  count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       5.411571      4.461296  ...    8.363584      5.258536
           std        3.939271      3.369910  ...    4.447250      2.962801
           min        0.000000      0.000000  ...    0.600000      0.000000
           25%        2.714525      2.170400  ...    5.092650      3.227700
           50%        4.864900      3.942950  ...    7.633350      4.770800
           75%        7.343500      6.005925  ...   10.795900      6.714700
           max       72.291600     51.805500  ...   23.942400     48.020800
AvgUtil_Th count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       5.357368      4.411557  ...    8.582809      5.281967
           std        3.859038      3.322752  ...    4.465378      2.959751
           min        0.000000      0.000000  ...    0.200000      0.000000
           25%        2.720650      2.145425  ...    5.427475      3.280900
           50%        4.756950      3.890550  ...    7.852200      4.783300
           75%        7.256550      5.955500  ...   10.864350      6.705500
           max       49.166600     48.250000  ...   25.202700     29.138700
AvgUtil_F  count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       5.560267      4.569286  ...    8.588034      5.471315
           std        4.044367      3.412811  ...    4.590126      3.103649
           min        0.000000      0.000000  ...    0.000000      0.000000
           25%        2.807600      2.222200  ...    5.423850      3.345800
           50%        4.979500      4.036150  ...    8.027250      4.925800
           75%        7.499050      6.134375  ...   10.901425      6.965800
           max       63.229100     38.680500  ...   27.530200     32.879200
AvgUtil_Sa count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       5.994033      5.021786  ...    9.543415      6.225789
           std        4.427123      3.822555  ...    4.866649      3.487585
           min        0.000000      0.000000  ...    0.000000      0.000000
           25%        2.909050      2.343700  ...    6.099500      3.836200
           50%        5.360200      4.444400  ...    8.645850      5.665000
           75%        8.209725      6.875525  ...   12.342075      7.932600
           max       44.444400     43.181800  ...   25.898700     37.685800
AvgUtil_Su count  11566.000000  11506.000000  ...  184.000000  20365.000000
           mean       7.061746      6.061953  ...   11.873145      7.804485
           std        5.211631      4.645072  ..

In [64]:
df1.loc[:,otl].describe(percentiles=[.01,.99]).loc[['min', '1%', '50%', '99%', 'max']]

Txn_Avg_Machine_Cnt  Room_Open_Hours  ...  AvgUtil_Su    nrpm
min                  2.0              8.0  ...    0.000000 -360.92
1%                   2.0              8.0  ...    0.000000    0.00
50%                  2.0             23.0  ...    7.345100   37.70
99%                 10.0             24.0  ...   23.175362  118.30
max                 42.0             24.0  ...   51.428500  802.51

[5 rows x 26 columns]

>0-3.01, >3.01-4.81, >4.81-7.01, >7.01-44.09

In [94]:
df1.OverallUtilBkt= pd.qcut(df1.Overall_Utilization, 4, labels=['Very Low', 'Low', 'Medium', 'High'])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:1: UserWarning: Pandas doesn't allow columns to be created via a new attribute name - see https://pandas.pydata.org/pandas-docs/stable/indexing.html#attribute-access
  """Entry point for launching an IPython kernel.


In [97]:
pd.crosstab(df1.OverallUtilBkt, [df1.vendor, df1.machine_type], df1.Txn_Avg_Machine_Cnt, aggfunc=('sum'))

vendor               CSCGo                    DI               
machine_type         DRYER STACKED WASHER  DRYER STACKED WASHER
Overall_Utilization                                            
Very Low              8118       0  10741   5641      31  12016
Low                   5772       0   7604  10339      52  17640
Medium                7315       0   7677  17433      99  18008
High                 11529       0   7023  27819     367  16612

In [ ]:
df1.groupby(['vendor', 'machine_type'])[sums].sum().T

In [77]:
df1.loc[df1.Full_Room_Util_Vend_PCT> 99].head()

Reporting_level  ... Days_To_ContractExpiration
959             Room  ...                      350.0
1232            Room  ...                      350.0
1742            Room  ...                    -1118.0
3377            Room  ...                     2145.0
4092            Room  ...                     2698.0

[5 rows x 40 columns]

In [81]:
df1.groupby(['vendor', 'machine_type'])[utils1].describe().T

vendor                                 CSCGo  ...            DI
machine_type                           DRYER  ...        WASHER
Overall_Utilization      count  11566.000000  ...  20365.000000
                         mean       5.196668  ...      4.938341
                         std        3.497400  ...      2.767383
                         min        0.070000  ...      0.040000
                         25%        2.480000  ...      3.010000
                         50%        4.670000  ...      4.450000
                         75%        7.240000  ...      6.310000
                         max       32.340000  ...     25.300000
Daily_Peak_Utilization   count  11566.000000  ...  20365.000000
                         mean      21.265506  ...     18.142143
                         std        7.712680  ...      6.590100
                         min        2.604163  ...      2.380952
                         25%       15.932291  ...     13.550724
                         50%       20.000000  ...     17.000000
                         75%       25.000000  ...     21.607143
                         max       50.000000  ...     50.000000
Full_Room_Util_Total_PCT count  11566.000000  ...  20365.000000
                         mean       0.463566  ...      1.047990
                         std        0.793241  ...      1.227523
                         min        0.000000  ...      0.000000
                         25%        0.020000  ...      0.120000
                         50%        0.130000  ...      0.610000
                         75%        0.560000  ...      1.600000
                         max       12.500000  ...     13.870000

[24 rows x 5 columns]

In [47]:
# for o in otl:
#   p1=df1.loc[:,o].describe(percentiles=[.01,.25,.50,.75,.99]).loc['1%']
#   p99= df1.loc[:,o].describe(percentiles=[.01,.25,.50,.75,.99]).loc['99%']
#   df1[o] = np.where(df1[o] <p1, p1,df1[o])
#   df1[o] = np.where(df1[o] >p99, p99,df1[o])

/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.
/usr/local/lib/python3.7/dist-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


In [44]:
df[o]

0         -370.0
1         1459.0
2         1934.0
3         2196.0
4         2166.0
           ...  
102029    1092.0
102030     -32.0
102031     709.0
102032    1274.0
102033    1310.0
Name: Days_To_ContractExpiration, Length: 94830, dtype: float64

In [18]:
utils=[]
for c in df.columns:
  if 'Util' in c:
    utils.append(c)

In [ ]:
utils

['Overall_Utilization',
 'Daily_Peak_Utilization',
 'Full_Room_Util_Gross_Hours',
 'Full_Room_Util_Vend_PCT',
 'Full_Room_Util_Total_PCT',
 'AvgUtil_M',
 'AvgUtil_T',
 'AvgUtil_W',
 'AvgUtil_Th',
 'AvgUtil_F',
 'AvgUtil_Sa',
 'AvgUtil_Su']

In [ ]:
utils= ['Overall_Utilization',
 'Daily_Peak_Utilization',
 'Full_Room_Util_Vend_PCT',
 'Full_Room_Util_Total_PCT',
 'AvgUtil_M',
 'AvgUtil_T',
 'AvgUtil_W',
 'AvgUtil_Th',
 'AvgUtil_F',
 'AvgUtil_Sa',
 'AvgUtil_Su']

In [ ]:
df1.columns

Index(['Reporting_level', 'vendor', 'as400id', 'location_id', 'Location_Name',
       'Branch_Id', 'Branch_Name', 'state', 'room_id', 'machine_type',
       'TimeSinceFirstVend', 'Tenure_Flag', 'Txn_Avg_Machine_Cnt',
       'Room_Open_Hours', 'Daily_Max_Room_Vend_Hours', 'Base_Vends',
       'Base_Vend_Amount', 'Top_Up_Vends', 'Top_Up_Vend_Amount',
       'Total_Vend_Hours', 'Total_Hours_Open', 'Overall_Utilization',
       'Max_Daily_Vends', 'Daily_Peak_Utilization',
       'Full_Room_Util_Gross_Hours', 'Full_Room_Util_Vend_PCT',
       'Full_Room_Util_Total_PCT', 'Avg_Daily_Vends', 'VDPM', 'Vend_Price',
       'AvgUtil_M', 'AvgUtil_T', 'AvgUtil_W', 'AvgUtil_Th', 'AvgUtil_F',
       'AvgUtil_Sa', 'AvgUtil_Su', 'nrpm', 'ContractExpirationDate',
       'Days_To_ContractExpiration'],
      dtype='object')

In [ ]:
sums= ['Txn_Avg_Machine_Cnt',
       'Room_Open_Hours', 'Daily_Max_Room_Vend_Hours', 'Base_Vends',
       'Base_Vend_Amount', 'Top_Up_Vends', 'Top_Up_Vend_Amount',
       'Total_Vend_Hours', 'Total_Hours_Open', 'Full_Room_Util_Gross_Hours']

In [ ]:
utils= ['Overall_Utilization',
 'Max_Daily_Vends',
 'Daily_Peak_Utilization',
 'Full_Room_Util_Vend_PCT',
 'Full_Room_Util_Total_PCT',
 'Avg_Daily_Vends',
 'VDPM',
 'Vend_Price',
 'AvgUtil_M',
 'AvgUtil_T',
 'AvgUtil_W',
 'AvgUtil_Th',
 'AvgUtil_F',
 'AvgUtil_Sa',
 'AvgUtil_Su']

In [ ]:
utils1= ['Overall_Utilization','Daily_Peak_Utilization','Full_Room_Util_Vend_PCT',
       'Full_Room_Util_Total_PCT']

In [ ]:
df1.groupby(['vendor', 'machine_type'])[sums].sum().T

vendor                             CSCGo  ...            DI
machine_type                       DRYER  ...        WASHER
Txn_Avg_Machine_Cnt         3.273400e+04  ...  6.427600e+04
Room_Open_Hours             2.494380e+05  ...  4.128520e+05
Daily_Max_Room_Vend_Hours   1.468395e+05  ...  2.218398e+05
Base_Vends                  4.628924e+06  ...  2.293427e+07
Base_Vend_Amount            7.355690e+06  ...  4.263262e+07
Top_Up_Vends                4.612340e+05  ...  0.000000e+00
Top_Up_Vend_Amount          2.349117e+05  ...  0.000000e+00
Total_Vend_Hours            5.371293e+06  ...  1.370238e+07
Total_Hours_Open            8.942614e+07  ...  2.386516e+08
Full_Room_Util_Gross_Hours  3.125745e+05  ...  1.669542e+06

[10 rows x 5 columns]

|\(&#39;index&#39;, &#39;&#39;)|\(&#39;CSCGo&#39;, &#39;DRYER&#39;)|\(&#39;CSCGo&#39;, &#39;WASHER&#39;)|\(&#39;DI&#39;, &#39;DRYER&#39;)|\(&#39;DI&#39;, &#39;STACKED&#39;)|\(&#39;DI&#39;, &#39;WASHER&#39;)|
|---|---|---|---|---|---|
|Txn\_Avg_Machine_Cnt|32734\.0|33045\.0|61232\.0|549\.0|64276\.0|
|Room\_Open_Hours|249438\.0|243271\.0|419631\.0|4122\.0|412852\.0|
|Daily\_Max_Room_Vend_Hours|146839\.464725|123910\.84767|270050\.828902|2791\.833299|221839\.827358|
|Base\_Vends|4628924\.0|5177790\.0|20357259\.0|294259\.0|22934272\.0|
|Base\_Vend_Amount|7355690\.19|8940723\.8|30783276\.52|512819\.15|42632617\.0|
|Top\_Up_Vends|461234\.0|107101\.0|0\.0|0\.0|0\.0|
|Top\_Up_Vend_Amount|234911\.73|66234\.93|0\.0|0\.0|0\.0|
|Total\_Vend_Hours|5371292\.96372|4290491\.378376|17073341\.301593|224972\.577354|13702375\.617953|
|Total\_Hours_Open|89426135\.0|89052572\.0|236529617\.0|2190806\.0|238651572\.0|
|Full\_Room_Util_Gross_Hours|312574\.5|352326\.0|1591777\.5|11296\.75|1669541\.5|

In [ ]:
df1.groupby(['vendor', 'machine_type'])[utils1].describe(percentiles=[0.05,.25,.5,.75,.95]).T

vendor                                 CSCGo  ...            DI
machine_type                           DRYER  ...        WASHER
Overall_Utilization      count  11566.000000  ...  20365.000000
                         mean       5.196668  ...      4.938341
                         std        3.497400  ...      2.767383
                         min        0.070000  ...      0.040000
                         5%         0.660000  ...      1.452000
                         25%        2.480000  ...      3.010000
                         50%        4.670000  ...      4.450000
                         75%        7.240000  ...      6.310000
                         95%       11.590000  ...     10.060000
                         max       32.340000  ...     25.300000
Daily_Peak_Utilization   count  11566.000000  ...  20365.000000
                         mean      21.265506  ...     18.142143
                         std        7.712680  ...      6.590100
                         min        2.604163  ...      2.380952
                         5%        10.869565  ...      9.417390
                         25%       15.932291  ...     13.550724
                         50%       20.000000  ...     17.000000
                         75%       25.000000  ...     21.607143
                         95%       34.817707  ...     30.620773
                         max       50.000000  ...     50.000000
Full_Room_Util_Vend_PCT  count  11566.000000  ...  20365.000000
                         mean       9.805348  ...     23.107523
                         std       12.460567  ...     21.630628
                         min        0.000000  ...      0.000000
                         5%         0.000000  ...      0.000000
                         25%        0.320000  ...      2.780000
                         50%        3.545000  ...     16.640000
                         75%       17.597500  ...     41.440000
                         95%       33.740000  ...     57.612000
                         max      100.000000  ...    100.000000
Full_Room_Util_Total_PCT count  11566.000000  ...  20365.000000
                         mean       0.463566  ...      1.047990
                         std        0.793241  ...      1.227523
                         min        0.000000  ...      0.000000
                         5%         0.000000  ...      0.000000
                         25%        0.020000  ...      0.120000
                         50%        0.130000  ...      0.610000
                         75%        0.560000  ...      1.600000
                         95%        2.050000  ...      3.440000
                         max       12.500000  ...     13.870000

[40 rows x 5 columns]

# CSCGo

In [ ]:
for u in utils:
  idx= df.loc[:,u]>100
  print(u, sum(idx))

Overall_Utilization 0
Daily_Peak_Utilization 0
Full_Room_Util_Vend_PCT 0
Full_Room_Util_Total_PCT 0
AvgUtil_M 0
AvgUtil_T 0
AvgUtil_W 0
AvgUtil_Th 0
AvgUtil_F 0
AvgUtil_Sa 0
AvgUtil_Su 1


In [ ]:
df1= df.loc[df.vendor=='CSCGo']

In [ ]:
df1.groupby('machine_type')[utils].describe(percentiles=[.05,.25,.5,.75,.95]).T.loc['Full_Room_Util_Vend_PCT']

machine_type         DRYER     STACKED        WASHER
count         15601.000000   49.000000  15716.000000
mean             35.026178   22.126939     42.765504
std              47.754827   46.559175     95.815908
min               0.000000    0.000000      0.000000
5%                0.000000    0.000000      0.000000
25%               0.710000    0.000000      1.357500
50%              10.180000    0.000000     14.735000
75%              66.050000    0.000000     90.212500
95%             124.760000  118.310000    133.330000
max             625.000000  190.910000   8666.670000

In [ ]:
df1.loc[df1.Full_Room_Util_Vend_PCT== 625].T

28735
Reporting_level                                                          Room
vendor                                                                  CSCGo
as400id                                                                708713
location_id                              284bc465-5e06-44a4-a4d2-6756d23fdca5
Location_Name                                          COMMONS OF MC LEAN THE
Branch_Id                                                                 7.0
Branch_Name                                                          Virginia
state                                                                      VA
room_id                     CG-284bc465-5e06-44a4-a4d2-6756d23fdca5-708713...
machine_type                                                            DRYER
TimeSinceFirstVend                                                        130
Tenure_Flag                                                           Tenured
Txn_Avg_Machine_Cnt                                                       1.0
Room_Open_Hours                                                             8
Daily_Max_Room_Vend_Hours                                                 1.0
Base_Vends                                                                  1
Base_Vend_Amount                                                         1.75
Top_Up_Vends                                                                0
Top_Up_Vend_Amount                                                        0.0
Total_Vend_Hours                                                          1.0
Total_Hours_Open                                                       1040.0
Overall_Utilization                                                       0.1
Max_Daily_Vends                                                             1
Daily_Peak_Utilization                                                   12.5
Full_Room_Util_Gross_Hours                                               6.25
Full_Room_Util_Vend_PCT                                                 625.0
Full_Room_Util_Total_PCT                                                  0.6
Avg_Daily_Vends                                                          0.01
VDPM                                                                   0.4095
Vend_Price                                                               1.75
AvgUtil_M                                                                 0.0
AvgUtil_T                                                                 0.0
AvgUtil_W                                                                 0.0
AvgUtil_Th                                                                0.0
AvgUtil_F                                                              0.6578
AvgUtil_Sa                                                                0.0
AvgUtil_Su                                                                0.0

In [ ]:
df1.loc[df1.Full_Room_Util_Vend_PCT > 8666].T

82798
Reporting_level                                                          Room
vendor                                                                  CSCGo
as400id                                                               5520213
location_id                              cadfdac1-cc9d-437c-9737-ea4b1e399314
Location_Name                                             MAIZLAND PROPERTIES
Branch_Id                                                                55.0
Branch_Name                                                             Tulsa
state                                                                      OK
room_id                     CG-cadfdac1-cc9d-437c-9737-ea4b1e399314-552021...
machine_type                                                           WASHER
TimeSinceFirstVend                                                        164
Tenure_Flag                                                           Tenured
Txn_Avg_Machine_Cnt                                                       1.0
Room_Open_Hours                                                             8
Daily_Max_Room_Vend_Hours                                                0.75
Base_Vends                                                                  0
Base_Vend_Amount                                                          0.0
Top_Up_Vends                                                                1
Top_Up_Vend_Amount                                                       0.25
Total_Vend_Hours                                                         0.75
Total_Hours_Open                                                       1312.0
Overall_Utilization                                                      0.06
Max_Daily_Vends                                                             1
Daily_Peak_Utilization                                                  9.375
Full_Room_Util_Gross_Hours                                               65.0
Full_Room_Util_Vend_PCT                                               8666.67
Full_Room_Util_Total_PCT                                                 4.95
Avg_Daily_Vends                                                          0.01
VDPM                                                                 0.046371
Vend_Price                                                                0.0
AvgUtil_M                                                                 0.0
AvgUtil_T                                                                 0.0
AvgUtil_W                                                                 0.0
AvgUtil_Th                                                                0.0
AvgUtil_F                                                                 0.0
AvgUtil_Sa                                                             0.3906
AvgUtil_Su                                                                0.0

In [ ]:
df.loc[df.Txn_Avg_Machine_Cnt==1, utils].describe()

Overall_Utilization  Daily_Peak_Utilization  ...    AvgUtil_Sa    AvgUtil_Su
count         42175.000000            42175.000000  ...  42175.000000  42175.000000
mean              4.939851               30.745106  ...      7.022237      8.009354
std               3.310869               13.298483  ...      5.269445      5.823244
min               0.020000                3.541663  ...      0.000000      0.000000
25%               2.610000               21.587300  ...      3.358300      3.863600
50%               4.360000               28.623187  ...      6.142100      7.095600
75%               6.590000               37.500000  ...      9.603950     11.037200
max              40.060000              162.500000  ...     91.354100    103.703700

[8 rows x 11 columns]

In [ ]:
df.loc[df.Txn_Avg_Machine_Cnt!=1, utils].describe()

Overall_Utilization  Daily_Peak_Utilization  ...    AvgUtil_Sa    AvgUtil_Su
count         54950.000000            54950.000000  ...  54950.000000  54950.000000
mean              6.345709               24.012390  ...      7.702590      9.340239
std               3.461763                8.921455  ...      4.363722      5.220341
min               0.020000                3.333333  ...      0.000000      0.000000
25%               3.900000               17.870369  ...      4.783600      5.767650
50%               5.860000               22.727273  ...      7.115300      8.759150
75%               8.240000               28.541667  ...      9.919800     12.223500
max              43.900000              118.298610  ...     60.156200     60.000000

[8 rows x 11 columns]